File: backend/app/models/medical_device.py
from sqlalchemy import Column, Integer, String, Text

from app.database import Base


class MedicalDevice(Base):
    __tablename__ = "medical_devices"

    id = Column(Integer, primary_key=True, index=True)

    device_test = Column(String(255), nullable=False)

    manufacturer = Column(String(255))

    approval_clearance_year = Column(String(255))

    regulatory_body = Column(String(255))

    country_region = Column(String(255))

    device_type = Column(String(255))

    primary_use = Column(Text)

    nafld_masld = Column(String(100))

    nash_mash = Column(String(100))

    fibrosis = Column(String(100))

    cirrhosis = Column(String(100))

File: frontend/src/types/medicalDevice.ts
export interface MedicalDevice {
  id: number;

  device_test: string | null;
  manufacturer: string | null;
  approval_clearance_year: string | null;
  regulatory_body: string | null;
  country_region: string | null;
  device_type: string | null;
  primary_use: string | null;
  nafld_masld: string | null;
  nash_mash: string | null;
  fibrosis: string | null;
  cirrhosis: string | null;
}

export interface MedicalDeviceListResponse {
  items: MedicalDevice[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
}